api/dbadapters/redshift.ts

```diff
@@ -77,27 +77,31 @@ export class RedshiftDbAdapter implements IDbAdapter {
     }
   }
 
-  public async evaluate(
-    queryOrAction: string | dataform.Table | dataform.Operation | dataform.Assertion
-  ) {
+  public async evaluate(queryOrAction: QueryOrAction, projectConfig?: dataform.ProjectConfig) {
     const validationQueries = collectEvaluationQueries(queryOrAction, false, (query: string) =>
       !!query ? `explain ${query}` : ""
-    );
+    ).map((validationQuery, index) => ({ index, validationQuery }));
+    const validationQueriesWithoutWrappers = collectEvaluationQueries(queryOrAction, false);
+
     const queryEvaluations = new Array<dataform.IQueryEvaluation>();
-    for (const { query, incremental } of validationQueries) {
+    for (const { index, validationQuery } of validationQueries) {
```

Comment:
This would be much faster if you run the queries in parallel, using Promise.all()

Reply:
Hm, that will make pre, post and the main operation of the query all race, shouldn't the order be preserved?